packages/astro/src/core/file/index.ts

```diff
@@ -0,0 +1,120 @@
+import type { AstroConfig } from '../../@types/astro';
+import { pathToFileURL, fileURLToPath } from 'node:url';
+import { normalizePath } from 'vite';
+import { prependForwardSlash } from '../path.js';
+import { unwrapId, viteID } from '../util.js';
+
+export type PathType =
+	/**
+	 * file:///Users/name/projects/todos/src/pages/index.astro
+	 */
```

Comment:
In addition to those comments, would it be possible to add a brief explanation as to when you should use each path types? This abstraction makes it easier to use the different ones, but I fear that without explanations, we'll still use the wrong ones at times.